src/python/pants/backend/debian/rules.py

```diff
@@ -36,24 +37,38 @@ async def package_debian_package(field_set: DebianPackageFieldSet) -> BuiltPacka
     dpkg_deb_path = await Get(
         BinaryPaths,
         BinaryPathRequest(
-            binary_name="touch",
+            binary_name="dpkg-deb",
             search_path=["/usr/bin"],
         ),
     )
     if not dpkg_deb_path.first_path:
-        raise EnvironmentError("Could not find the `touch` program on search paths ")
+        raise EnvironmentError("Could not find the `dpkg-deb` program on search paths ")
 
-    output_filename = field_set.output_path.value_or_default(field_set.address, file_ending="deb")
+    output_filename = field_set.output_path.value_or_default(
+        field_set.address,
+        file_ending="deb",
+    )
+
+    # TODO: need to create a digest from the directory from the "sources" field of
+    #  "debian_package" target, i.e. field_set.control.filespec['includes'][0]
+    package_metadata_dir_digest = await Get(Digest, CreateDigest("?"))
```

Comment:
@Eric-Arellano so I need to create a Digest for a directory (that contains `DEBIAN/control` file):

```
debian_package(
    name='sample-debian-package',
    description='Useful tools installed as Debian package.',
    packages=[':files'],
    sources=['sample-debian-package'],
    symlinks={"myapp": "pex-target-name"},
    install_prefix="/opt/mycompany/tools",
    tags=["tooling"],
)
```

Do I need https://www.pantsbuild.org/docs/rules-api-file-system#pathglobs?

Reply:
The `control` file is read from disk, right? You are not creating it dynamically? You'll use this: https://www.pantsbuild.org/docs/rules-api-and-target-api#the-sources-field

You'll use `hydrated_sources = await Get(HydratedSources, HydrateSourcesRequest(field_set.control))`, which works because `DebianControlFile` subclasses `Sources`.

Then, you can access `hydrated_sources.snapshot.digest` and put it as the `input_digest` to the `Process`.